src/components/PopupWithForm.js

```diff
@@ -24,4 +37,14 @@ export default class PopupWithForm extends Popup {
       this._popupForm.reset();
```

Comment:
Please, delete `this._popupForm.reset();` from line 37

You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.